BleWinrtDll/BleWinrtDll.cpp

```diff
@@ -561,60 +559,201 @@ bool SendData(BLEData* data, bool block) {
 	return result;
 }
 
-fire_and_forget ReadDataAsync(BLECharacteristic* id, BLEData* data, condition_variable* signal, bool* result) {
+//Internal class for handling a coroutine constantly Reading data from characteristic
+//Thread safe?
+class Reading {
```

Comment:
Does this class need to be in `BleWinrtDll.cpp`? Or can it be moved to a separate file to keep the size of this script low?

Reply:
I would suggest if we move this out we should also sepparate out a few more bits like helper functions etc.
It can be moved, however I've kept it all in one file to keep with the style of the cloned project (although braking it by using oop).